app/scripts/main/controllers/url-selector.controller.js

```diff
@@ -0,0 +1,107 @@
+/*jshint -W003, -W098, -W033 */
+(function () {
+  'use strict';
+
+  /**
+   * @ngdoc function
+   * @name ngAmrsApp.controller:MainCtrl
+   * @description
+   * # MainCtrl
+   * Controller of the ngAmrsApp
+   */
+  angular
+    .module('ngAmrsApp')
+    .controller('UrlSelectorCtrl', UrlSelectorCtrl);
+  UrlSelectorCtrl.$nject = ['$rootScope', '$scope', 'OpenmrsSettings', 'EtlRestServicesSettings', 'AuthService', '$state'];
+
+  function UrlSelectorCtrl($rootScope, $scope, OpenmrsSettings, EtlRestServicesSettings, AuthService, $state) {
+
+    $scope.currentMode = '';
+
+    $scope.currentRestBaseUrl = '';
+
+    $scope.currentEtlRestBaseUrl = '';
+
+    $scope.urlRestBaseList = [];
+    $scope.urlEtlRestBaseList = [];
+
+    $scope.urlToAdd = '';
```

Comment:
Likewise here to avoid confusion with etl it would be nice to explicitly name Openmrs things with Openmrs in them. This could be opemrsUrlToAdd. This comment applies to other similar variables related to Openmrs so I won't add more comments about them in subsequent code.